### PR TITLE
Add method for creating assets

### PR DESCRIPTION
**Pull request by classiemilio**
Some random changes snuck in cause of my prettify plugin for VSCode

### CODE REVIEW

File: src/CAL.js [hunk 1]
```diff
@@ -1,45 +1,44 @@
...
 
   async getEntriesForContentType(contentType) {
     const allEntries = {};
-    debug('Fetching all entries...');
+    debug("Fetching all entries...");
     try {
-      const entries = await this.client.getEntries(contentType);;
-      debug('All entries fetched.');
+      const entries = await this.client.getEntries(contentType);
+      console.log(entries);
```
Comment (gwitwer) [inline src/CAL.js]: extra console log. prefer debug if we want to keep it

File: src/CAL.js [hunk 2]
```diff
@@ -1,45 +1,44 @@
-const path = require('path');
-const yaml = require('js-yaml');
-const { PassThrough } = require('stream');
+const path = require("path");
```
Comment (gwitwer) [inline src/CAL.js]: `" --> '` everywhere

File: src/Client.js [hunk 3]
```diff
@@ -97,19 +98,48 @@ class Client {
...
+      client.createEntryWithId(...args);
     const throttledCreateEntry = throttle(createEntryWrapper);
     const newEntry = await throttledCreateEntry(contentType, id, entry);
     await newEntry.publish();
     return newEntry;
   }
+
+  async createAssetFromUrl(url, title) {
+    const client = await this.client;
+    const createAssetWrapper = async (...args) => client.createAsset(...args);
```
Comment (gwitwer) [inline src/Client.js]: nice! thanks for following pattern. I tried to abstract this flow into a "throttle wrapper builder" but it didn't work like i expected 🤷‍♂

File: src/Client.js [hunk 4]
```diff
@@ -97,19 +98,48 @@ class Client {
...
+    const throttledCreateAsset = throttle(createAssetWrapper);
+
+    const imageFetch = await fetch(url);
+    const buffer = await imageFetch.buffer();
+    const fileType = await FileType.fromBuffer(buffer);
+
+    let asset = await throttledCreateAsset({
```
Comment (gwitwer) [inline src/Client.js]: maybe overkill at this point but we may want to abstract the `object --> Asset` logic into a method

File: src/Client.js [hunk 5]
```diff
@@ -97,19 +98,48 @@ class Client {
...
-    const id = makeHash(entry.fields.key['en-US']);
+    const id = makeHash(entry.fields.key["en-US"]);
     console.log(id);
     if (overwrite) {
       await removeEntry(id);
     }
 
     const client = await this.client;
-    const createEntryWrapper = async (...args) => client.createEntryWithId(...args);
+    const createEntryWrapper = async (...args) =>
```
Comment (gwitwer) [inline src/Client.js]: If we split this into two lines, let's use parens to wrap.

Comment (classiemilio) [reply src/Client.js]: Prettier isn't letting me add these haha...

File: src/Client.js (hunk 3 above)
Comment (classiemilio) [reply src/Client.js]: It works nicely!

File: src/Client.js (hunk 4 above)
Comment (classiemilio) [reply src/Client.js]: Good call, done.

File: src/Client.js (hunk 5 above)
Comment (gwitwer) [reply src/Client.js]: 😕 o well

**Pull request merged**